addons/purchase_stock/models/stock.py

```diff
@@ -132,18 +137,25 @@ def _compute_qty_to_order_computed(self):
     def _compute_lead_days(self):
         return super()._compute_lead_days()
 
+    @api.onchange('supplier_id')
```

Comment:
You can also do it in the write/create method if modifying it in a compute/inverse is too hacky